docs/extend/modular-resources/create.md

```diff
@@ -0,0 +1,515 @@
+
+A configured *module* can make one or more *modular resources* available for configuration.
+
+{{% /alert %}}
+
+## Create a custom modular resource
+
+To create your own modular resource, code a module in Go or Python using the module support libraries provided by [Viam's SDKs](/program/apis/) that implements at least one new model or type of {{< glossary_tooltip term_id="resource" text="resource" >}}:
+
+{{< tabs >}}
+{{% tab name="Define a New Model of a Built-In Resource Type" %}}
+
+1. [Code a new resource model](#code-a-new-resource-model) implementing all methods the Viam RDK requires in the API definition of its built-in type (ex. `rdk:component:base`).
+Import your custom model and API into the main program and register the new resource model with your chosen SDK's helper methods.
+
+2. [Code a main program](#code-a-main-program-that-serves-as-the-entry-point-file-to-the-module) that starts the module after adding your desired resources from the registry.
+This main program is the "entry point" to your module.
+
+3. [Compile or package](#compile-the-module-into-an-executable) the module into a single executable that can receive a socket argument from Viam, open the socket, and start the module at the entry point.
+
+{{% /tab %}}
+{{% tab name="Define a New Type of Resource" %}}
+
+1. Define the messages and methods of the new API in [protobuf](https://github.com/protocolbuffers/protobuf), then [generate code](https://grpc.io/docs/languages/python/generated-code/) in Python or Go and use the generated code to implement the higher level server and client functions required.
+
+2. [Code at least one model](#code-a-new-resource-model) of this new resource.
+Make sure to implement every method required in your API definition.
+Import your custom models and APIs into the main program and register the new resource models with your chosen SDK's helper methods.
+
+3. [Code a main program](#code-a-main-program-that-serves-as-the-entry-point-file-to-the-module) that starts the module after adding your desired resources from the registry.
+This main program is the "entry point" to your module.
+
+4. [Compile or package](#compile-the-module-into-an-executable) the module into a single executable that can receive a socket argument from Viam, open the socket, and start the module at the entry point.
+
+{{% /tab %}}
+{{% /tabs %}}
+
+### Code a new resource model
+
+{{% alert title="Note" color="note" %}}
+
+The following example module registers a modular resource implementing Viam's built-in [Base API](/components/base/#api) [(rdk:service:base)](/extend/modular-resources/key-concepts/#models) as a new model, `"mybase"`:
+
+The Go code for the custom model (<file>mybase.go</file>) and module entry point file (<file>main.go</file>) is adapted from the full demo modules available on the [Viam GitHub](https://github.com/viamrobotics/rdk/blob/main/examples/customresources).
+
+{{% /alert %}}
+
+[Go Client SDK](https://pkg.go.dev/go.viam.com/rdk):
```

Comment:
I think the format of this section is a bit difficult to parse visually because our expanders have bold text, whereas the surrounding "headings" aren't headings so the Python vs Go and the file names get lost visually. Maybe try using tabs like with the existing doc, and/or maybe put in horizontal lines to break up the sections? Or use headings or something? Something to emphasize the desired hierarchy/sections instead of the expanders being the biggest boldest thing.

Reply:
I agree. I tried to put the code in tabs but the expanders don't work inside of tabs and the code is chunky. Unsure @JessamyT or @npentrel do you have any ideas on this? maybe the lines?

Reply:
TBH I found the old format with the tabs and no expanders easier to follow/look at. Yeah I'd play with the "---" line separators perhaps. Maybe I'll mess around locally and make a suggestion if I come up with anything I actually like.

Reply:
Ooh another thing: since we do Python as the front tab, then Go, on most other pages, maybe do that here?

Reply:
Okay so, if you want to stick with this general organizational structure (as opposed to the way it was before) but use tabs, you could use tabs with the "details" tag like https://gist.github.com/pierrejoubert73/902cc94d79424356a8d20be2b382e1ab and those do indeed nest within tabs.

Reply:
Also, I'm thinking you don't really need the link to the Go or Python SDKs on these "headings"

Reply:
With details tags, it looks like this: 
![image](https://github.com/viamrobotics/docs/assets/75634662/56fdae9d-dc13-4d1d-b723-3c450939ec36)
Here's a gist incorporating all my feedback on this section (does not include executable and next steps):
https://gist.github.com/JessamyT/514cd1da1694ade1bdff0a96a57bcda0